logic/src/commonMain/kotlin/com/wire/kalium/logic/data/conversation/MLSConversationRepository.kt

```diff
@@ -277,4 +304,65 @@ class MLSConversationDataSource(
         conversationDAO.getAllMembers(conversationID).first().map { idMapper.fromDaoModel(it.user) }
     }
 
+    private suspend fun executeOperation(groupID: GroupID, operation: suspend () -> Either<CoreFailure, Unit>) =
+        operation()
+            .flatMapLeft {
+                handleCommitFailure(it, groupID, operation)
+            }
+
+    private suspend fun handleCommitFailure(
```

Comment:
a suggestion for this part of the code:
I think this part would grow more and more in the future, and it's not clean IMO.
My suggestion: to have this functions and the retry mechanism inside the Failure sealed classes, and then here we only call the retry once for all! based on the sealed type the correct function will be invoked.
[if this is not clear, we can have a call to discuss it]

Reply:
Let's have a call